scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -780,10 +829,38 @@ class ContainerChecker( StoppableThread ):
         Monitors containers to check when they start and stop running.
     """
 
-    def __init__( self, config, global_config, logger, socket_file, docker_api_version, host_hostname, data_path, log_path,
+    def __init__( self, config, global_config, logger, socket_file, docker_api_version, agent_pod, host_hostname, log_path,
                   include_all, include_controller_info, namespaces_to_ignore,
                   ignore_pod_sandboxes ):
+        """
 
+        :param config: The configuration for the monitor which includes options for the container checker
+        :param global_config: The global configuration file
+        :param logger: The logger instance to use
+        :param socket_file: The docker socket file
+        :param docker_api_version: The API version to use for docker
+        :param agent_pod: The pod name and namespace for the agent
+        :param host_hostname: The hostname for this node
+        :param log_path: The path to the container logs
+        :param include_all: Whether or not include all container by default
+        :param include_controller_info: Whether or not to include the controller information for all pods upload
+        :param namespaces_to_ignore: The namespaces whose pods should not be uploaded
+        :param ignore_pod_sandboxes: Whether or not to recognize pod sandboxes
+
+        :type config: dict
+        :type global_config: Configuration
+        :type logger: Logger
+        :type socket_file: str
+        :type docker_api_version: str
+        :type agent_pod: QualifiedName
+        :type host_hostname: str
+        :type log_path: str
+        :type include_all: bool
+        :type include_controller_info: bool
+        :type namespaces_to_ignore: [str]
+        :type ignore_pod_sandboxes: bool
+        """
+        StoppableThread.__init__(self, name='container checker thread')
```

Comment:
Note, I added this init to the super class.  If someone thinks that's not the right thing to do, let me know.

Reply:
I think the correct thing is probably to have `ContainerChecker` inherit from `object`.  This is likely an artifact of a refactoring, but the main thread of ContainerChecker now has a `StoppableThread` as a member and it's that thread which does all the checking.  As far as I can tell, it never uses its inherited `StoppableThread` functionality.

Probably best to remove this (and inherit from object), but leaving it in doesn't look like it will cause harm.

Reply:
@echee2 Ok, we probably should remove this then.  I'm sorry, that's my bad.